integration-tests/runtime-tests/src/generators/crowdloanGenerators/crowdloanRewardGenerator.ts

```diff
@@ -0,0 +1,105 @@
+import { ApiPromise, Keyring, WsProvider } from '@polkadot/api';
+import { KeyringPair } from '@polkadot/keyring/types';
+import minimist from 'minimist';
+import * as R from 'ramda';
+import { args } from '../../utils/args';
+
+/**
+ * Contains all generation methods as static async functions
+ * 
+ * WARN: await Delay can cause problems!
+**/
+export class crowdloanRewardGenerator {
+  /**
+   * 
+   * @param { ApiPromise } api Connected API Promise.
+   * @param { KeyringPair } sudoKey with sudo access.
+   * @param { KeyringPair } walletAlice 
+   */
+  public static async testCrowdloanRewards(api: ApiPromise, sudoKey: KeyringPair, walletAlice: KeyringPair) {
+    /*
+      Populate
+      Initialize
+      Associate (automatically execute first claim)
+    */
+
+    const toHexString = bytes =>
+      Array.prototype.map.call(bytes, x => ('0' + (x & 0xFF).toString(16)).slice(-2)).join('');
+
+    const delay = () => new Promise(res => setTimeout(res, 24_000));
+
+    const vesting48weeks = 100800;
+    const accounts =
+      R.unfold(n => n > 100 ? false : [[
+        { RelayChain: walletAlice.derive("/contributor-" + n.toString()).publicKey },
+        n * 1_000_000_000_000,
+        vesting48weeks
+      ], n + 1], 1);
+
+    const populateHash =
+      await api.tx.sudo.sudo(
+        api.tx.crowdloanRewards.populate(accounts)
+      ).signAndSend(sudoKey, { nonce: -1 });
+    console.debug('Populated crowdloan with hash: ', populateHash.toHex());
+
+    await delay();
+
+    const initializeHash =
+      await api.tx.sudo.sudo(
+        api.tx.crowdloanRewards.initialize()
+      ).signAndSend(sudoKey, { nonce: -1 });
+
+    console.debug('Initialized crowdloan with hash: ', initializeHash.toHex());
+
+    await delay();
```

Comment:
Not a blocker for this PR, but would be better to await for either a duration, or for an event related to the previous tx to be emitted.

Reply:
You're right, definitely a better solution. As mentioned above, this file will be reworked to wait for results instead of using delay, similar to the "txCrowdloanRewardsTests.ts" file.